addons/account/models/account_move.py

```diff
@@ -1065,6 +1064,8 @@ def _compute_bank_partner_id(self):
                 move.bank_partner_id = move.commercial_partner_id
             else:
                 move.bank_partner_id = move.company_id.partner_id
+            filtred_bank_partner_id = move.bank_partner_id.filtered(lambda bank: bank.company_id.id in (False, move.company_id.id))
+            move.invoice_partner_bank_id = filtred_bank_partner_id.bank_ids and filtred_bank_partner_id.bank_ids[0]
```

Comment:
* Don't compute fields that are not supposed to be computed
* Don't change it if it was already a valid bank
* You can also use `filtred_bank_partner_id.bank_ids[:1]` to take the first record safely
* filt***e***red